library/find_unused_disk_module.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/python
+
+DOCUMENTATION = '''
+---
+module: find_unused_disk_module
+short_description: Get first unused disk
+description:
+    - Disks are considered in ascending alphanumeric sorted orderself.
+    - First disk to meet all conditions is considered 'empty' and returned (using kernel device name).
+        - 1. No known signatures exist on the disk, with the exception of partition tables.
+        - 2. If there is a partition table on the disk, it contains no partitions.
+        - 3. The disk has no holders to eliminate the possibility of it being a multipath or dmraid member device.
+        - 4. Device can be opened with exclusive access to make sure no other software is using it.
+    - If no disks meet all criteria, "Unable to find empty disk" will be returned.
+author: Eda Zhou (@edamamez)
+'''
+
+EXAMPLES = '''
+- name: test finding first empty device module
+  hosts: localhost
+  tasks:
+    - name: run module
+      find_unused_disk_module:
+      register: testout
+    - name: dump test output
+      debug:
+        msg: '{{ testout }}'
+'''
+
+RETURN = '''
+disk_name:
+    description: Information about first empty disk
+    returned: On success
+    type: complex
+    samples: "/dev/sda"
+             "Unable to find empty disk"
+'''
+
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils import facts
+
+
+# Ensure that no known signatures exist on the disk, with the exception of 
+# partition tables. (blkid -p <device>)
+def no_signature(run_command, disk):
+    signatures = run_command(['blkid', '-p', disk])
+    return not 'UUID' in signatures[1]
+
+
+# Ensure that the disk has no holders to eliminate the possibility of it being a
+#  multipath or dmraid member device. (/sys/class/block/<device>/holders/ empty)
+def no_holders(run_command, disk_name):
+    holders = run_command(['ls', '/sys/class/block/' + disk_name + '/holders/'])
```

Comment:
Use `os.listdir`[1] instead of `ls' here. In general, https://docs.python.org is your friend.

[1] https://docs.python.org/3/library/os.html#os.listdir